metricflow/model/validations/unique_valid_name.py

```diff
@@ -57,64 +56,125 @@ def check_valid_name(name: str, data_source_name: Optional[str] = None) -> List[
 
     @staticmethod
     @validate_safely(whats_being_done="checking data source sub element names are unique")
-    def _validate_data_source_elements(data_source: DataSource) -> List[ValidationIssue]:
-        issues = []
-        name_and_type_tuples: List[Tuple[ElementReference, str]] = []
+    def _validate_data_source_elements(data_source: DataSource) -> List[ValidationIssueType]:
+        issues: List[ValidationIssueType] = []
+        name_and_type_tuples: List[Tuple[ElementReference, str, ContextType]] = []
```

Comment:
I feel like I'm on book 4 of the increasingly inappropriately named Hitchiker's Guide to the Galaxy trilogy....

It seems `name_and_type_tuples` would benefit from a new name.